### PR TITLE
Adding precommit hook for linting

### PR DESCRIPTION
**Pull request by alexose**
I think this is a good compromise between linting on rebuild (too often) and only linting during CI (not enough).

### CODE REVIEW

File: script/build.js [hunk 1]
```diff
@@ -22,8 +22,22 @@ const webpack = require('metalsmith-webpack');
...
+const fs = require('fs');
+
 const sourceDir = '../content/pages';
 
+// Make sure git pre-commit hooks are installed
+['pre-commit'].forEach(hook => {
+  const src = `hooks/${hook}`;
+  const dest = `.git/hooks/${hook}`;
+  if (fs.existsSync(src)) {
+    if (!fs.existsSync(dest)) {
```
Comment (jbalboni) [inline script/build.js]: I wonder if there's any problem with copying it every build? With this check any changes to the pre-commit hook won't get installed for devs who've already built once.

Comment (alexose) [inline script/build.js]: That would be fine, I think.  There's probably a more graceful way to do this...

Comment (jbalboni) [inline script/build.js]: There are tools to manage what hooks are installed, but I like the simplicity of this.

**Comment by jbalboni** [comment]
LGTM

**Comment by alexose** [comment]
This isn't ready to merge yet due to the insane amount of linting errors for tests and such

**Comment by jbalboni** [comment]
@alexose It's weird that you're getting linting errors for tests. Those should be linted at the moment.

**Comment by alexose** [comment]
Hm, maybe it's a configuration issue.  Let me look a deeper into it.

**Pull request merged**